src/logic/addressBook/store/reducer/addressBook.ts

```diff
@@ -18,13 +19,19 @@ export const buildAddressBook = (storedAddressBook: AddressBookState): AddressBo
   })
 }
 
-export default handleActions(
+type AddressBookPayload = { addressBook: AddressBookState }
+type EntryPayload = { entry: AddressBookEntry }
+type RemoveEntryPayload = { entryAddress: string }
+
+type Payloads = AddressBookPayload | EntryPayload | RemoveEntryPayload
+
+export default handleActions<AppReduxState['addressBook'], Payloads>(
   {
-    [LOAD_ADDRESS_BOOK]: (state, action) => {
+    [LOAD_ADDRESS_BOOK]: (state, action: Action<AddressBookPayload>) => {
```

Comment:
It's possible to type `state` too?

Reply:
It's already typed: https://github.com/gnosis/safe-react/blob/fix/add-types-for-redux-action/src/logic/addressBook/store/reducer/addressBook.ts#L28 (`AppReduxState['addressBook']`)